repository_service_tuf_worker/repository.py

```diff
@@ -1171,39 +1525,69 @@ def _run_online_roles_bump(self, force: Optional[bool] = False):
                 f"{Targets.type} don't use online key, skipping 'Targets' role"
             )
         else:
-            targets: Metadata = self._storage_backend.get(Targets.type)
+            targets: Metadata[Targets] = self._storage_backend.get(
+                Targets.type
+            )
             if force or (targets.signed.expires - today) < timedelta(
                 hours=self._hours_before_expire
             ):
                 self._update_targets_delegations_key(targets)
                 self._bump_and_persist(targets, Targets.type)
                 logging.info("Bumped version of 'Targets' role")
-                snapshot_bump = True
 
         if force:
             # Updating all delegated target roles.
-            timestamp = self._update_timestamp(
-                self._update_snapshot(bump_all=True)
+            self._update_timestamp(
+                self._update_snapshot(bump_all=True, only_target=True)
             )
-            snapshot_bump = True
             logging.info("Targets and delegated Targets roles version bumped")
         else:
             # Updating only those delegated roles that have expired.
+            targets: Metadata[Targets] = self._storage_backend.get(
+                Targets.type
+            )
             delegated_roles: List[str] = []
-            delegated_names = self._settings.get_fresh("DELEGATED_ROLES_NAMES")
-            for role in delegated_names:
-                role_md: Metadata[Targets] = self._storage_backend.get(role)
+            if targets.signed.delegations.succinct_roles:
+                s_roles = targets.signed.delegations.succinct_roles.get_roles()
+                delegated_roles = [r for r in s_roles]
+            else:
+                delegated_roles = list(targets.signed.delegations.roles.keys())
+
+            for role in delegated_roles:
+                if targets_crud.read_role_deactivated_by_rolename(
+                    self._db, role
+                ):
+                    delegated_roles.remove(role)
+                    logging.debug(f"Role '{role}' is deactivated, skipping")
+                    continue
```

Comment:
Maybe directly read from database the active roles?

Reply:
We can discuss it.
I'm trying to avoid making the DB a source of trust. In that case, we use Metadata.
I think this is a good point for us to revisit.

Reply:
https://github.com/repository-service-tuf/repository-service-tuf-worker/issues/595